### DISCUSSION TITLE
Parameter type coersion failing with Binding Error

### DISCUSSION THREAD

**Discussion by amirrajan** [duckdb/duckdb#3670] [Q&A]
I'm using the node bindings for duckdb. 

If I pass in `[1, 2]` for parameters and use the following sql, I get a Binding Error from Duck:

```sql
select ni + $1 p02
from (
  values
    ('a', 1::float8, 1::int),
    ('b', 2::float8, 2::int),
    ('c', 3::float8, 3::int)
) as T(ch, nf, ni)
where nf in ($1, $2)
```

Would there be a way to have numeric parameters automatically coerce to respective column types?

**Comment by Mytherin** [comment]
We can probably defer binding in this case as well, I will have a look.
